tools/python/openlr/quality.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import xml.etree.ElementTree as ET
+import operator
+
+from collections import namedtuple
+from itertools import islice
+from math import sin, cos, atan2, radians, sqrt
+
+LatLon = namedtuple('LatLon', 'lat, lon')
+
+def distance(x, y):
+    """Implements https://en.wikipedia.org/wiki/Haversine_formula.
+
+    >>> int(distance(LatLon(55.747043, 37.655554), LatLon(55.754892, 37.657013)))
+    875
+    >>> int(distance(LatLon(60.013918, 29.718361), LatLon(59.951572, 30.205536)))
+    27910
+    """
+
+    φ1, φ2 = map(radians, [x.lat, y.lat])
+    λ1, λ2 = map(radians, [x.lon, y.lon])
+    Δφ = φ2 - φ1
+    Δλ = λ2 - λ1
+    a = sin(Δφ/2)**2 + cos(φ1) * cos(φ2) * sin(Δλ/2)**2
+    R = 6356863
+    return 2 * R * atan2(sqrt(a), sqrt(1 - a))
+
+def lcs(l1, l2, eq=operator.eq):
+    """Finds the longest common sequence of l1 and l2.
+    Returns a list of common parts and a list of differences.
+
+    >>> lcs([1, 2, 3], [2])
+    ([2], [1, 3])
+    >>> lcs([1, 2, 3, 3, 4], [2, 3, 4, 5])
+    ([2, 3, 4], [1, 3, 5])
+    >>> lcs('banana', 'baraban')
+    (['b', 'a', 'a', 'n'], ['a', 'r', 'b', 'n', 'a'])
+    >>> lcs('abraban', 'banana')
+    (['b', 'a', 'a', 'n'], ['a', 'r', 'n', 'b', 'a'])
+    >>> lcs([1, 2, 3], [4, 5])
+    ([], [4, 5, 1, 2, 3])
+    >>> lcs([4, 5], [1, 2, 3])
+    ([], [1, 2, 3, 4, 5])
+    """
+    prefs_len = [
+        [0] * (len(l2) + 1)
+        for _ in range(len(l1) + 1)
+    ]
+    for i in range(1, len(l1) + 1):
+        for j in range(1, len(l2) + 1):
+            if eq(l1[i - 1], l2[j - 1]):
+                prefs_len[i][j] = prefs_len[i - 1][j - 1] + 1
+            else:
+                prefs_len[i][j] = max(prefs_len[i - 1][j], prefs_len[i][j - 1])
+    common = []
+    diff = []
+    while i and j:
```

Comment:
Could you please explicitly initialize i and j before the loop?

Reply:
Why? It's python: i and j are just the same as in the last for iteration.
Only def and class introduce a new scope in python.

Reply:
Ok, done.